server/src/downloaders/photo-downloader.ts

```diff
@@ -1,3 +1,19 @@
+import http from "http";
+import https from "https";
+import Stream from "stream";
+import fs from "fs";
+
 export const downloadPhotoWithCache = ({url, cacheDir}: {url: string; cacheDir: string}) => {
-  // TODO implement
+    if (fs.existsSync(cacheDir)) return;
+
+    const protocol = url.indexOf("https") === 0 ? https : http;
```

Comment:
This should be replaced with
```
// Import fs like this to make fs functions use promises
import fs from 'fs/promise';

// ...

const response = await fetch();
await fs.writeFile(join(cacheDir, await response.arrayBuffer()))
```

It does the same thing, but the node-fetch interface works browser and server side (And is cleaner).